spring-boot-sample-project/src/main/java/com/example/demo/controller/UserController.java

```diff
@@ -0,0 +1,53 @@
+package com.example.demo.controller;
+
+import com.example.demo.pojo.User;
+import com.example.demo.service.UserService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import javax.validation.Valid;
+import java.util.List;
+import java.util.Optional;
+
+@RestController
+@RequestMapping("/")
+public class UserController {
+
+    @Autowired
+    private UserService userService;
+
+    @GetMapping("/user")
+    public ResponseEntity<List<User>> getAllUser(){
+        List<User> userList = userService.getAllUser();
+        return ResponseEntity.ok(userList);
+    }
+
+    @GetMapping("/user/{name}")
```

Comment:
this should be query param within `/user` API endpoint
Ideally detailed call for a resource should be using a pk/identifier eg. id
`user/{id}` as rest API is meaningful than `user/{name}`